src/screens/CreatePost/elements/UserProfile.js

```diff
@@ -107,16 +109,27 @@ const UserProfile = ({
         <ToggleSwitch
           value={isAnonymous}
           onValueChange={() => setTypeUser(!isAnonymous)}
-          labelLeft="Anonymity"
+          labelLeft="Incognito"
           backgroundActive={COLORS.lightgrey}
           backgroundInactive={COLORS.lightgrey}
           circleInActiveColor={COLORS.signed_primary}
           inactiveTextColor={COLORS.signed_primary}
           styleLabelLeft={styles.switch}
+          isDisabled={isToggleDisabled}
         />
       </View>
     </>
   );
 };
 
+UserProfile.propTypes = {
+  setTypeUser: PropTypes.func,
+  username: PropTypes.string,
+  photo: PropTypes.string,
+  onPress: PropTypes.func,
+  isAnonymous: PropTypes.bool,
+  anonUserInfo: PropTypes.object,
+  isToggleDisabled: PropTypes.bool
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The PropTypes have been updated to include `isToggleDisabled`. This is good for type checking in development mode.
```diff
+  isToggleDisabled: PropTypes.bool
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->